src/main/java/io/pinecone/commons/IndexInterface.java

```diff
@@ -38,13 +38,13 @@ default UpsertRequest validateUpsertRequest(List<VectorWithUnsignedIndices> vect
         for (VectorWithUnsignedIndices vectorWithUnsignedIndices : vectorWithUnsignedIndicesList) {
             SparseValuesWithUnsignedIndices sparseValuesWithUnsignedIndices = vectorWithUnsignedIndices.getSparseValuesWithUnsignedIndices();
 
-            Vector vector = buildUpsertVector(vectorWithUnsignedIndices.getId(),
-                    vectorWithUnsignedIndices.getValuesList(),
-                    sparseValuesWithUnsignedIndices.getIndicesWithUnsigned32IntList(),
-                    sparseValuesWithUnsignedIndices.getValuesList(),
-                    vectorWithUnsignedIndices.getMetadata());
+                Vector vector = buildUpsertVector(vectorWithUnsignedIndices.getId(),
+                        vectorWithUnsignedIndices.getValuesList(),
+                        (sparseValuesWithUnsignedIndices != null) ? sparseValuesWithUnsignedIndices.getIndicesWithUnsigned32IntList() : null,
+                        (sparseValuesWithUnsignedIndices != null) ? sparseValuesWithUnsignedIndices.getValuesList() : null,
```

Comment:
We were getting `NullPointerExceptions` here if the user had passed `null` for these values in the `VectorWithUnsignedIndices` class.